sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -7,16 +7,27 @@
 import com.azure.storage.blob.models.ParallelTransferOptions;
 import com.azure.storage.blob.specialized.BlockBlobAsyncClient;
 
-import java.util.regex.Pattern;
+import java.net.MalformedURLException;
+import java.net.URL;
 
 /**
  * This class provides helper methods for common model patterns.
  *
  * RESERVED FOR INTERNAL USE.
  */
 public class ModelHelper {
-    public static final Pattern IP_V4_URL_PATTERN = Pattern
-        .compile("(?:\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})|(?:localhost)");
+
+    /**
+     * Determines whether or not the passed authority is IP style, that is it is of the format
```

Comment:
Doesnt matter too much but the wording here is a little weird "that is it is", should there be a comma somewhere or something?